screeps-game-api/src/game.rs

```diff
@@ -241,7 +245,24 @@ pub fn time() -> u32 {
 
 /// See [http://docs.screeps.com/api/#Game.getObjectById]
 ///
+/// This gets an object expecting a specific type and will return a `ConversionError` if the type
+/// does not match.
+///
+/// If all you want to assume is that something has an ID, use [`get_object_erased`].
+/// [http://docs.screeps.com/api/#Game.getObjectById]: http://docs.screeps.com/api/#Game.getObjectById
+pub fn get_object_typed<T>(id: &str) -> Result<Option<T>, ConversionError>
```

Comment:
Not necessary for this PR (I can put it with `HasId`), but would you like having a wrapper type for IDs?